app/models/exercise.rb

```diff
@@ -29,6 +40,10 @@ def side?
     !core
   end
 
+  def unlocked_by_user?(user)
+    solutions.find_by(user: user).present?
```

Comment:
This will be faster too as it won't retrieve the record from the db or have to create any AR objects:
`solutions.where(user_id: user.id).exists?`